File: deamhome/src/main/java/sosteam/deamhome/domain/category/entity/ItemDetailCategory.java
package sosteam.deamhome.domain.category.entity;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import sosteam.deamhome.domain.item.entity.Item;
import sosteam.deamhome.global.entity.BaseEntity;

import java.util.List;

@Document
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ItemDetailCategory extends BaseEntity {

    @NotNull
    private String title;

    @DBRef(lazy = true)
    private List<Item> items;

    @DBRef(lazy = true)
    private List<ItemCategory> itemCategories;
}

File: src/main/kotlin/sosteam/deamhome/domain/category/entity/ItemDetailCategory.kt
package sosteam.deamhome.domain.category.entity

import lombok.Builder
import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.mapping.Document
import sosteam.deamhome.global.entity.EmbeddedEntity

@Document
@Builder
data class ItemDetailCategory(
	@Indexed(unique = true)
	var title: String,
	var itemIdList: MutableList<String> = mutableListOf()
) : EmbeddedEntity() {
	fun modifyItems(itemIdList: MutableList<String>): MutableList<String> {
		this.itemIdList = itemIdList
		return this.itemIdList
	}

}